pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -2489,8 +2489,16 @@ default 'raise'
             # We can avoid going through pydatetime paths, which is robust
             #  to datetimes outside of pydatetime range.
             ts = _TSObject()
-            check_dts_bounds(&dts, self._creso)
-            ts.value = npy_datetimestruct_to_datetime(self._creso, &dts)
+            try:
+                ts.value = npy_datetimestruct_to_datetime(self._creso, &dts)
+            except OverflowError as err:
+                # TODO: create shared function to create this format from dts struct
```

Comment:
might be worth doing this as a preliminary.  id be much more comfortable merging that as a standalone without the C stuff

Reply:
Fair enough. I just took care of that here because its pretty small, but can split out if a deal breaker

Reply:
Not a deal breaker, just easier for me to merge promptly